pkg/patterns/declarative/reconciler.go

```diff
@@ -403,3 +411,30 @@ func (r *Reconciler) IsKustomizeOptionUsed() bool {
 func (r *Reconciler) SetSink(sink Sink) {
 	r.options.sink = sink
 }
+
+func parseListKind(infos *manifest.Objects) error{
+		for i, item := range infos.Items{
+			if item.Kind == "List" {
+				fmt.Println("haaa!")
+				obj := item.UnstructuredObject()
+
+				err := obj.EachListItem(func(obj runtime.Object) error{
+					castItem := obj.(*unstructured.Unstructured)
+					newObj, err := manifest.NewObject(castItem)
+					if err != nil {
+						return err
+					}
+					infos.Items = append(infos.Items, newObj)
+					return nil
+				})
+
+				if err != nil {
+					return err
+				}
+
+				infos.Items = append(infos.Items[:i], infos.Items[i+1:]...)
```

Comment:
I think leigh was suggesting that it might be easier to just keep a second list:

```
var out []*manifest.Object

if IsList(obj) {
  out = append(out, extract(obj))
} else {
  out = append(out, obj)
}
return manifest.NewManifest(out)
}
```

I'm not sure what happens to the iteration when you concurrently modify `infos`!